src/Ansatz/Chain.jl

```diff
@@ -227,6 +227,30 @@ function Base.rand(rng::Random.AbstractRNG, sampler::ChainSampler, ::Type{Open},
     Chain(Operator(), Open(), arrays)
 end
 
+Tenet.contract(tn::Chain, query::Symbol, args...; kwargs...) =
+    Tenet.contract!(copy(tn)::Chain, query, args...; kwargs...)
+Tenet.contract!(tn::Chain, query::Symbol, args...; kwargs...) =
```

Comment:
You need to add `@valsplit` to dispatch based on `Val` values.
```suggestion
Tenet.contract(tn::Chain, query::Symbol, args...; kwargs...) =
    Tenet.contract!(copy(tn)::Chain, query, args...; kwargs...)
@valsplit 2 Tenet.contract!(tn::Chain, query::Symbol, args...; kwargs...) =
```

Reply:
I tried that before but I get `UndefVarError: `kwargs` not defined`, have you seen this before?